app/code/Magento/MediaGallery/Model/Asset/Command/DeleteByDirectoryPath.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGallery\Model\Asset\Command;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\Exception\CouldNotDeleteException;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\MediaGalleryApi\Model\Asset\Command\DeleteByDirectoryPathInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class DeleteByDirectoryPath
+ *
+ * Remove asset(s) that correspond the provided directory path
+ */
+class DeleteByDirectoryPath implements DeleteByDirectoryPathInterface
+{
+    private const TABLE_MEDIA_GALLERY_ASSET = 'media_gallery_asset';
+
+    private const MEDIA_GALLERY_ASSET_PATH = 'path';
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * DeleteById constructor.
+     *
+     * @param ResourceConnection $resourceConnection
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        LoggerInterface $logger
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Delete media asset(s) by path
+     *
+     * @param string $directoryPath
+     *
+     * @return void
+     * @throws CouldNotDeleteException
+     */
+    public function execute(string $directoryPath): void
+    {
+        try {
+            $this->validateDirectoryPath($directoryPath);
```

Comment:
`validateDirectoryPath` throws a `LocalizedException`. I don't think it's necessary to recapture that exception, I propose to move the call out of try-catch block (also it may be good to change the `LocalizedException` to more specific `CouldNotDeleteException` in `validateDirectoryPath`)